### ISSUE TITLE
Multi graphics card training model

### ISSUE THREAD

**Issue by monkeycc** [ultralytics/ultralytics#20766]
```
GPU H20-NVLink(96GB) * 4  
CPU  64 vCPU AMD EPYC 9K84 96-Core Processor
Memory 600GB,

Ultralytics 8.3.143 🚀 Python-3.12.3 torch-2.5.1+cu124 CUDA:0 (NVIDIA H20, 97383MiB)
                                                       CUDA:1 (NVIDIA H20, 97383MiB)
                                                       CUDA:2 (NVIDIA H20, 97383MiB)
                                                       CUDA:3 (NVIDIA H20, 97383MiB)
```

```

+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 565.57.01              Driver Version: 565.57.01      CUDA Version: 12.7     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA H20                     On  |   00000000:34:00.0 Off |                  Off |
| N/A   46C    P0            252W /  500W |    9049MiB /  97871MiB |    100%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   1  NVIDIA H20                     On  |   00000000:43:00.0 Off |                  Off |
| N/A   39C    P0            121W /  500W |    6558MiB /  97871MiB |    100%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   2  NVIDIA H20                     On  |   00000000:83:00.0 Off |                  Off |
| N/A   55C    P0            204W /  500W |    9039MiB /  97871MiB |    100%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   3  NVIDIA H20                     On  |   00000000:A4:00.0 Off |                  Off |
| N/A   43C    P0            190W /  500W |    8557MiB /  97871MiB |    100%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+

```

`imgsz=640 , batch = 60`

```
Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
1/100      2.25G       1.09      2.221      1.072         47        640: 100%|██████████| 392/392 [02:03<00:00,  3.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95):  18%|█▊        | 60/339 [01:11<05:30,  1.18s/it]
```

 GPU_mem 2.25G

I feel the speed is slow
GPU_mam 2.25GB is also unknown if it is normal

**Comment by UltralyticsAssistant** [comment]
👋 Hello @monkeycc, thank you for sharing your setup and performance details with Ultralytics 🚀! This is an automated response to help you get started—an Ultralytics engineer will also assist you soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate further.

If your question relates to custom training or performance, please include:
- As much detail as possible about your training setup (config, command used, logs, dataset samples).
- Confirmation that you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For new users, we recommend reviewing the [Docs](https://docs.ultralytics.com/) for many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples.

Join the Ultralytics community where it suits you best:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) to share and learn from others

## Upgrade

Ensure you are running the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in any of these up-to-date, verified environments (all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks again for your report! Please update this issue with the requested details so we can assist you faster 🛠️

**Comment by Y-T-G** [comment]
Training occurs on multiple GPUs. But validation only runs on single GPU.
